content/influxdb/v2.2/api-guide/iot-starter/python.md

```diff
@@ -0,0 +1,743 @@
+level directory.
+In `config.ini`, paste the following:
+
+```ini
+[APP]
+INFLUX_URL = {{INFLUX_URL}}
+INFLUX_TOKEN = {{INFLUX_TOKEN}}
+INFLUX_ORG = {{INFLUX_ORG_ID}}
+INFLUX_BUCKET = {{INFLUX_BUCKET_FOR_TELEMETRY}}
+INFLUX_BUCKET_AUTH = {{INFLUX_BUCKET_FOR_DEVICES}}
+```
+{{% caption %}}[@influxdata/iot-python-app config.ini](https://github.com/influxdata/iot-python-api/blob/main/config.ini){{% /caption %}}
+
+The following is a sample configuration.
+```ini
+[APP]
+INFLUX_URL = https://us-west-2-2.aws.cloud2.influxdata.com/
+INFLUX_TOKEN = 52Pc_ZkJsRh1PKzlwrK8yO6jWSDh6WPAHbfqp-5aROz4zBnY2mvkKws9YoYzksGH3_Xp90rVqo2PRiajTxaUcw==
+INFLUX_ORG = bea7ea952287f70d
+INFLUX_BUCKET = sly's Bucket
+INFLUX_BUCKET_AUTH = devices_auth
+```
+
+
+## Create IoT Virtual Device
+The IoT virtual device generates simulated weather data for you to store in
+InfluxDB. 
+
+In the `./api` directory of your project, create a `sensor.py` file.
+
+[//]: # Probably preferable to link the file rather than have the whole file written up()
+```python
+import json
+import random
+import urllib3
+
+
+http = urllib3.PoolManager()
+
+
+# Helper function to fetch lat lon data
+def fetch_json(url):
+    """Fetch JSON from url."""
+    response = http.request('GET', url)
+    if not 200 <= response.status <= 299:
+        raise Exception(f'[HTTP - {response.status}]: {response.reason}')
+    config_fresh = json.loads(response.data.decode('utf-8'))
+    return config_fresh
+
+
+class Sensor:
+    def __init__(self):
+        self.id = ''
+        self.temperature = None
+        self.pressure = None
+        self.humidity = None
+        self.geo = None
+
+    def generate_measurement(self):
+        return round(random.uniform(0, 100))
+
+    def geo(self):
+        """
+        Get GEO location from https://freegeoip.app/json/'.
+        :return: Returns a dictionary with `latitude` and `longitude` key.
+        """
+        try:
+            return fetch_json('https://freegeoip.app/json/')
+        except Exception:
+            return {
+                'latitude':  self.generate_measurement(),
+                'longitude':  self.generate_measurement(),
+            }
+```
+{{% caption %}}[@influxdata/iot-python-app ./api/sensor.py](https://github.com/influxdata/iot-python-api/blob/main/api/sensor.py){{% /caption %}}
+
+The Sensor object's function `generate_measurement()` will be used to simulate 
+weather data.
+
+
+## Develop with the API
+
+## Store Virtual IoT Device to InfluxDB
+
+In this section, you will learn how to use the python client library to store
+virtual device information in InfluxDB.
+Virtual device information will be stored in your first bucket. 
+Telemetry data from the device will be stored in the second bucket. 
+(You will learn more about this later on in the guide).
+
+In your `./api` directory, create a `devices.py` file. This file holds the 
+core functionality for your app.
+
```

Comment:
Here you show the whole `devices.py` file up front. I think it would be better to walk through adding each piece of the file and explain what each does. Then at the end of this section, show the end result (the entire file).